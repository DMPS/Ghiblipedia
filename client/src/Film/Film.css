.film {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  border-radius: 20px;
  max-width: 400px;
  text-align: center;
  padding: 8px 16px;
  margin: 10px;
  top: 0;
  transition: all 0.1s ease-in;
  display: flex;
  flex-direction: column;
}

.film > .title {
  opacity: 0.85;
}

.film > .title:hover {
  opacity: 1;
}

a {
  color: inherit;
}

.film > .text {
  color: black;
  flex-grow: 1;
}

.filmmakers {
  font-size: 12px;
  font-weight: bold;
  color: #999;
  text-transform: uppercase;
  letter-spacing: 0.05em;
  margin: 2em 0 2em 0;
  margin-bottom: auto;
}

.film:hover {
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23);
  margin-top: 27px;
  margin-bottom: 27px;
}

.container {
  padding: 2px 16px;
}
