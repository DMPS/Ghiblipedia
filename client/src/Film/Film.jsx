import React, { Component } from "react";
import "./Film.css";
import Button from "../Button/Button";

class Film extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    const { director, title, producer, description } = this.props;
    return (
      <div className="film">
        <h1 className="title">
          <a
            target="_blank"
            href={`http://www.google.com/search?q=${encodeURIComponent(title)}`}
          >
            {title}
          </a>
        </h1>
        <p className="text">
          <i>{description}</i>
        </p>
        <Button
          text={`I NEED MORE ${title.toUpperCase()}! ðŸ¤¤`}
          onClick={() => {
            console.log("clicked");
          }}
        />
        <span className="filmmakers">
          {director === producer
            ? `Directed and produced by ${director}`
            : `Directed by ${director} and produced by ${producer}`}
        </span>
      </div>
    );
  }
}

export default Film;
