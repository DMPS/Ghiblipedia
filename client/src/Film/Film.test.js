import React from "react";
import Film from "./Film";
import renderer from "react-test-renderer";

describe("Film", () => {
  it("renders without crashing", () => {
    const tree = renderer
      .create(
        <Film director="blah" producer="blah" title="blah" description="blah" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("shows both director and producer ", () => {
    const tree = renderer
      .create(
        <Film
          director="The Director"
          producer="The Producer"
          title="Test Film"
          description="The Description"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("doesn't repeat the director/producer's name ", () => {
    const tree = renderer
      .create(
        <Film
          director="The Director"
          producer="The Director"
          title="Test Film"
          description="The Description"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
