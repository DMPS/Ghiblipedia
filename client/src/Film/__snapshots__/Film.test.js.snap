// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film doesn't repeat the director/producer's name  1`] = `
<div
  className="film"
>
  <h1
    className="title"
  >
    <a
      href="http://www.google.com/search?q=Test%20Film"
      target="_blank"
    >
      Test Film
    </a>
  </h1>
  <p
    className="text"
  >
    <i>
      The Description
    </i>
  </p>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      I NEED MORE TEST FILM! ðŸ¤¤
    </span>
  </button>
  <span
    className="filmmakers"
  >
    Directed and produced by The Director
  </span>
</div>
`;

exports[`Film renders without crashing 1`] = `
<div
  className="film"
>
  <h1
    className="title"
  >
    <a
      href="http://www.google.com/search?q=blah"
      target="_blank"
    >
      blah
    </a>
  </h1>
  <p
    className="text"
  >
    <i>
      blah
    </i>
  </p>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      I NEED MORE BLAH! ðŸ¤¤
    </span>
  </button>
  <span
    className="filmmakers"
  >
    Directed and produced by blah
  </span>
</div>
`;

exports[`Film shows both director and producer  1`] = `
<div
  className="film"
>
  <h1
    className="title"
  >
    <a
      href="http://www.google.com/search?q=Test%20Film"
      target="_blank"
    >
      Test Film
    </a>
  </h1>
  <p
    className="text"
  >
    <i>
      The Description
    </i>
  </p>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      I NEED MORE TEST FILM! ðŸ¤¤
    </span>
  </button>
  <span
    className="filmmakers"
  >
    Directed by The Director and produced by The Producer
  </span>
</div>
`;
